src/all/nijiero/src/eu/kanade/tachiyomi/extension/all/nijiero/Nijiero.kt

```diff
@@ -0,0 +1,165 @@
+package eu.kanade.tachiyomi.extension.all.nijiero
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.model.UpdateStrategy
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.select.Evaluator
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class Nijiero : HttpSource() {
+    override val name = "Nijiero"
+    override val lang = "all"
+    override val supportsLatest = false
+
+    override val baseUrl = "https://www.nijiero-ch.com"
+
+    override val client = network.client.newBuilder().followRedirects(true).build()
+    class TagFilter : Filter.Select<String>("Category", nijieroTags)
+
+    override fun pageListParse(response: Response): List<Page> {
+        val document = response.asJsoup()
+        return document.select("#entry > ul > li a[href]").mapIndexed { i, linkElement ->
+            val linkUrl = linkElement.attr("href").removeSuffix(".webp").removePrefix(baseUrl)
+            Page(i, linkUrl, linkUrl)
```

Comment:
`Page`'s `url` should be the document where the image is found, not a link to the image itself.

Reply:
wdym? it returns the image...

Reply:
2nd argument is the `url`, 3rd argument is the `imageUrl`. If anything you should set the 2nd argument to `document.location()`.